cudax/include/cuda/experimental/__container/vector.cuh

```diff
@@ -0,0 +1,1585 @@
+  _LIBCUDACXX_REQUIRES(__properties_match<_OtherProperties...>)
+  explicit vector(const vector<_Tp, _OtherProperties...>& __other)
+      : __buf_(__other.get_resource(), __other.__size_)
+      , __size_(__other.__size_)
+  {
+    if (__other.__size_ != 0)
+    {
+      this->__copy_cross<const_pointer, __transfer_kind<_OtherProperties...>>(
+        __other.__unwrapped_begin(), __other.__unwrapped_end(), __unwrapped_begin(), __other.__size_);
+    }
+  }
+
+  //! @brief Move-constructs from a vector with matching properties
+  //! @param __other The other vector.
+  _LIBCUDACXX_TEMPLATE(class... _OtherProperties)
+  _LIBCUDACXX_REQUIRES(__properties_match<_OtherProperties...>)
+  explicit vector(vector<_Tp, _OtherProperties...>&& __other) noexcept
+      : __buf_(_CUDA_VSTD::move(__other.__buf_))
+      , __size_(_CUDA_VSTD::exchange(__other.__size_, 0))
+  {}
+
+  //! @brief Constructs an empty vector using a memory resource
+  //! @param __mr The memory resource to allocate the vector with.
+  //! @note No memory is allocated.
+  vector(__resource __mr)
+      : __buf_(_CUDA_VSTD::move(__mr), 0)
+      , __size_(0)
+  {}
+
+  //! @brief Constructs a vector of size \p __size using a memory resource and value-initializes \p __size elements
+  //! @param __mr The memory resource to allocate the vector with.
+  //! @param __size The size of the vector. Defaults to zero
+  //! @note If `__size == 0` then no memory is allocated.
+  explicit vector(__resource __mr, const size_type __size)
+      : __buf_(_CUDA_VSTD::move(__mr), __size)
+      , __size_(__size)
+  {
+    if (__size != 0)
+    {
+      this->__value_initialize_n(__unwrapped_begin(), __size);
+    }
+  }
+
+  //! @brief Constructs a vector of size \p __size using a memory resource and copy-constructs \p __size elements from
+  //! \p __value
+  //! @param __mr The memory resource to allocate the vector with.
+  //! @param __size The size of the vector.
+  //! @param __value The value all elements are copied from.
+  //! @note If `__size == 0` then no memory is allocated.
+  explicit vector(__resource __mr, const size_type __size, const _Tp& __value)
+      : __buf_(_CUDA_VSTD::move(__mr), __size)
+      , __size_(__size)
+  {
+    if (__size != 0)
+    {
+      this->__fill_n(__unwrapped_begin(), __size, __value);
+    }
+  }
+
+  //! @brief Constructs a vector of size \p __size using a memory and leaves all elements uninitialized
+  //! @param __mr The memory resource to allocate the vector with.
+  //! @param __size The size of the vector.
+  //! @warning This constructor does *NOT* initialize any elements. It is the user's responsibility to ensure that the
+  //! elements within `[vec.begin(), vec.end())` are properly initialized, e.g with `cuda::std::uninitialized_copy`.
+  //! At the destruction of the \c vector all elements in the range `[vec.begin(), vec.end())` will be destroyed.
+  explicit vector(__resource __mr, const size_type __size, ::cuda::experimental::uninit_t)
```

Comment:
is there a reason why `uninit` is the last argument? tags for overload selection are usually the first argument.

Reply:
The reason is that it is symmetric to the other `resource` + `size` constructors.

Not a hill I would die for